### ISSUE TITLE
Validation needed when expanding the U height of a device type

### ISSUE THREAD

**Issue by LordBoBCUP** [digitalocean/netbox#466]
I am getting a weird error when trying to view one of the racks I had previously setup. I have recently upgraded to v1.5.2 and when viewing a specific rack for a specific site I am getting an error, traceback below. 
Traceback:

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/exception.py" in inner
  39.             response = get_response(request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _legacy_get_response
  249.             response = self._get_response(request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/opt/netbox-1.5.1/netbox/dcim/views.py" in rack
  217.         'front_elevation': rack.get_front_elevation(),

File "/opt/netbox-1.5.1/netbox/dcim/models.py" in get_front_elevation
  422.         return self.get_rack_units(face=RACK_FACE_FRONT, remove_redundant=True)

File "/opt/netbox-1.5.1/netbox/dcim/models.py" in get_rack_units
  412.                     elevation[device.position]['device'] = device

Exception Type: KeyError at /dcim/racks/11/
Exception Value: 20

**Comment by jeremystretch** [comment]
v1.5.2 hasn't been released yet. Are you running the development branch?

**Comment by LordBoBCUP** [comment]
Sorry Jeremy, running v1.5.1 from the master branch. Must have fat fingered it.

**Comment by jeremystretch** [comment]
I'm not able to replicate this. Can you provide a list of devices installed in the rack, including their positions/faces and device type heights?

**Comment by LordBoBCUP** [comment]
I think I have narrowed it down. I had made a change to a device type, and not viewed that particular rack since making the change. The device I had changed is only in that 1 rack as well which is why I didn't see the error anywhere else. Sadly I have changed a few other things in that rack (positions within the rack) but I think it was because once I had changed it from 1RU to 2RU there was another 1RU device in the second RU spot for the first device if that makes sense. 

I have just retested expanding the 1RU device type to 2RU and viewing the rack is fine. However, if you expand a 1RU device type into a 2RU device and on a rack where the next RU is already consumed, you appear to get that error. Perhaps we need some validation when changing the RU of a device type to see all the racks where its used, if the next RU slot is free or not? I know changing the device type post using it isn't recommended, but in this instance, the specs says its 1RU but really, its fatter than 1RU and the second RU slot is not really usable so I wanted to effectively mark it 2RU or at least the second RU occupied to avoid people trying to use it in the future.
